base/commands/project/project_list_templates.go

```diff
@@ -0,0 +1,234 @@
+package project
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/paths"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/store"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/log"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/serialization"
+)
+
+type ListCmd struct{}
+
+const (
+	flagLocal            = "local"
+	flagForce            = "force"
+	nextFetchTimeKey     = "project.templates.nextFetchTime"
+	templatesKey         = "project.templates"
+	cacheRefreshInterval = 10 * time.Minute
+)
+
+var storeFolder = filepath.Join("caches", "templates")
+
+type Template struct {
+	Name   string `json:"name"`
+	Source string
+}
+
+func (lc ListCmd) Init(cc plug.InitContext) error {
+	cc.SetPositionalArgCount(0, 0)
+	cc.SetCommandUsage("list-templates [flags]")
+	cc.AddBoolFlag(flagForce, "", false, false, "force to list most recent from remote")
+	cc.AddBoolFlag(flagLocal, "", false, false, "list the templates which exist on local environment")
+	help := "Lists templates that can be used while creating projects."
+	cc.SetCommandHelp(help, help)
+	return nil
+}
+
+func (lc ListCmd) Exec(ctx context.Context, ec plug.ExecContext) error {
+	isLocal := ec.Props().GetBool(flagLocal)
+	isForce := ec.Props().GetBool(flagForce)
+	if isLocal && isForce {
+		return fmt.Errorf("%s and %s flags cannot be set at the same time", flagForce, flagLocal)
+	}
+	ts, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
+		sp.SetText(fmt.Sprintf("Listing templates"))
+		return listTemplates(ec.Logger(), isLocal, isForce)
+	})
+	if err != nil {
+		return err
+	}
+	tss := ts.([]Template)
+	if len(tss) == 0 {
+		ec.PrintlnUnnecessary("No templates found")
+	}
+	rows := make([]output.Row, len(tss))
+	for i, t := range tss {
+		rows[i] = output.Row{
+			output.Column{
+				Name:  "Template Source",
+				Type:  serialization.TypeString,
+				Value: t.Source,
+			},
+			output.Column{
+				Name:  "Template Name",
+				Type:  serialization.TypeString,
+				Value: t.Name,
+			},
+		}
+	}
+	stop()
+	return ec.AddOutputRows(ctx, rows...)
+}
+
+func listTemplates(logger log.Logger, isLocal bool, isForce bool) ([]Template, error) {
+	sa := store.NewStoreAccessor(paths.Join(paths.Home(), storeFolder), logger)
+	if isLocal {
+		return listLocalTemplates()
+	}
+	if fetch, err := isFetch(sa); err != nil {
```

Comment:
Can we continue and list the templates if `isFetch` returns an error?
Logging the error in DEBUG would be good.